dev/Repeater/APITests/Common/DataAsElementElementFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Windows.UI.Xaml.Controls;
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests.RepeaterTests.Common
+{
+    using Microsoft.UI.Xaml.Controls;
+    using ElementFactory = Microsoft.UI.Xaml.Controls.ElementFactory;
+
+    class DataAsElementElementFactory : ElementFactory
+    {
+        protected override UIElement GetElementCore(ElementFactoryGetArgs args)
+        {
+            var element = args.Data as UIElement;
+            if (element != null)
+            {
+                return element;
+            }
+
+            var button = new Button();
+            button.Content = args.Data;
+            return button;
```

Comment:
Is this used ? If not, perhaps just throw if it is not a UIElement.

Reply:
I'd just return nullptr if i were you.

---
In reply to: [355558439](https://github.com/microsoft/microsoft-ui-xaml/pull/1740#discussion_r355558439) [](ancestors = 355558439)